packages/flutter/test/material/tabs_test.dart

```diff
@@ -3950,6 +3950,141 @@ void main() {
+    int color = 0x00000000;
+    MaterialStateColor buildMSC() {
+      final Color selectedColor = Color(color++);
+      final Color unselectedColor = Color(color++);
+      return MaterialStateColor
+          .resolveWith((Set<MaterialState> states) {
+            if (states.contains(MaterialState.selected)) {
+              return selectedColor;
+            }
+            return unselectedColor;
+          });
+    }
+
+    final Color materialLabelColor = buildMSC();
+    final Color labelColor = Color(color++);
+    final Color unselectedLabelColor = Color(color++);
+
+    // this is to make sure labelStyles (in TabBar and in TabBarTheme) don't
+    // affect label's color. for details: https://github.com/flutter/flutter/pull/109541#issuecomment-1294241417
+    final TextStyle labelStyle = TextStyle(color: Color(color++));
+    final TextStyle unselectedLabelStyle = TextStyle(color: Color(color++));
+
+    final TabBarTheme materialTabBarTheme = TabBarTheme(
+      labelColor: buildMSC(),
+      unselectedLabelColor: Color(color++),
+      labelStyle: TextStyle(color: buildMSC()),
+      unselectedLabelStyle: TextStyle(color: Color(color++)),
+    );
+    final TabBarTheme tabBarTheme = TabBarTheme(
+      labelColor: Color(color++),
+      unselectedLabelColor: Color(color++),
+      labelStyle: TextStyle(color: Color(color++)),
+      unselectedLabelStyle: TextStyle(color: Color(color++)),
+    );
+    final TabBarTheme tabBarThemeWithNullUnselectedLabelColor = TabBarTheme(
+      labelColor: Color(color++),
+      labelStyle: TextStyle(color: Color(color++)),
+      unselectedLabelStyle: TextStyle(color: Color(color++)),
+    );
+
+    Future<void> buildTabBar({
+      bool isLabelColorMSC = false,
+      bool isLabelColorNull = false,
+      bool isUnselectedLabelColorNull = false,
+      bool isTabBarThemeMSC = false,
+      bool isTabBarThemeNull = false,
+      bool isTabBarThemeUnselectedLabelColorNull = false,
+    }) async {
+      final TabBarTheme? effectiveTheme = isTabBarThemeNull
+          ? null : isTabBarThemeUnselectedLabelColorNull
+            ? tabBarThemeWithNullUnselectedLabelColor
+            : isTabBarThemeMSC
+              ? materialTabBarTheme
+              : tabBarTheme;
+      await tester.pumpWidget(boilerplate(
+        child: Theme(
+          data: ThemeData(tabBarTheme: effectiveTheme),
+          child: DefaultTabController(
+            length: 2,
+            child: TabBar(
+              labelColor: isLabelColorNull ? null : isLabelColorMSC ? materialLabelColor : labelColor,
+              unselectedLabelColor: isUnselectedLabelColorNull ? null : unselectedLabelColor,
+              labelStyle: labelStyle,
+              unselectedLabelStyle: unselectedLabelStyle,
+              tabs: const <Widget>[Text('1'), Text('2')],
+            ),
+          ),
+        ),
+      ));
+    }
+
+    Color? getTab1Color() => IconTheme.of(tester.element(find.text('1'))).color;
```

Comment:
Comparing colors for == can be tricky because of MaterialColor etc. Could defend against that (future problem) by returning `.color.value` here and below. Then:
```dart
expect(getTab1Color(), getSelectedColor(materialLabelColor))
```

Reply:
ohh. i didn't think of that. thanks!